apm-commons/apm-util/src/main/java/org/apache/skywalking/apm/util/ConfigInitializer.java

```diff
@@ -50,43 +53,45 @@ private static void initNextLevel(Properties properties, Class<?> recentConfigTy
                 String configKey = (parentDesc + "." + field.getName()).toLowerCase();
                 Class<?> type = field.getType();
 
-                if (type.equals(Map.class)) {
+                if (Map.class.isAssignableFrom(type)) {
                     /*
-                     * Map config format is, config_key[map_key]=map_value
-                     * Such as plugin.opgroup.resttemplate.rule[abc]=/url/path
+                     * Map config format is, config_key[map_key]=map_value, such as plugin.opgroup.resttemplate.rule[abc]=/url/path
+                     * "config_key[]=" will generate a bleak Map , user could use this mechanism to set a blank Map
                      */
-                    // Deduct two generic types of the map
                     ParameterizedType genericType = (ParameterizedType) field.getGenericType();
                     Type[] argumentTypes = genericType.getActualTypeArguments();
-
-                    Type keyType = null;
-                    Type valueType = null;
-                    if (argumentTypes != null && argumentTypes.length == 2) {
-                        // Get key type and value type of the map
-                        keyType = argumentTypes[0];
-                        valueType = argumentTypes[1];
+                    Type keyType = argumentTypes[0];
+                    Type valueType = argumentTypes[1];
+                    // A chance to set a blank map
+                    if (properties.containsKey(configKey + "[]")) {
+                        field.set(null, initBlankMap(type));
+                    } else {
+                        // Set the map from config key and properties
+                        Map map = readMapType(type, configKey, properties, keyType, valueType);
+                        if (map.size() != 0) {
+                            field.set(null, map);
+                        }
                     }
-                    Map map = (Map) field.get(null);
-                    // Set the map from config key and properties
-                    setForMapType(configKey, map, properties, keyType, valueType);
-                } else if (Collection.class.isAssignableFrom(type)) {
+                    continue;
+                }
+                if (!properties.containsKey(configKey)) {
+                    continue;
+                }
+                //In order to guarantee user could override the default value to blank , we parse the value even if it's blank
```

Comment:
I don't think `override xxx to` is correct grammar. You could say `use xxx to override` or `xx is reset as yyy`

Reply:
In order to guarantee the default value could be reset as empty/blank , we parse the value even if it's blank